### ISSUE TITLE
Don't generate Autocomplete for hide commands

### ISSUE THREAD

**Issue by power721** [remkop/picocli#1076]
I want to hide some commands.
It's hide from help message, but it does not hide from generated Autocomplete.

**Comment by remkop** [comment]
Thank you for raising this!
That makes sense. 

Will you be able to provide a pull request for this?

**Comment by remkop** [comment]
@power721, what version of picocli are you using?
Looking at the code for `AutoComplete`, hidden commands should be skipped...
Can you show me an example that reproduces this issue?

**Comment by power721** [comment]
I use 4.3.2.
```xml
        <dependency>
            <groupId>info.picocli</groupId>
            <artifactId>picocli</artifactId>
            <version>4.3.2</version>
        </dependency>
```
I hide the `item get` command:
```java
    @Command(name = "get", hidden = true, description = "Get item from the Lockbox")
    public void get(@Parameters(description = "The key", paramLabel = "KEY") String key) {
        System.out.println(lockbox.get(key));
    }
```

Example output:
```bash
linux:~ # ./lockbox item
contains  create    get       remove    set       update
linux:~ # ./lockbox item
Missing required subcommand
Usage: lockbox item [COMMAND]
Manage lockbox items
Commands:
  contains  Check item exist in the Lockbox
  create    Create item in the Lockbox
  remove    Remove item from the Lockbox
  set       Set item in the Lockbox
  update    Update item in the Lockbox

```

**Comment by remkop** [comment]
Fixed in master.
This will be included in the upcoming picocli 4.4 release.

Thank you again for raising this!
